src/System.Management/src/System/Management/ManagementScope.cs

```diff
@@ -0,0 +1,1687 @@
+        ///    </code>
+        ///    <code lang='VB'>Dim s As New ManagementScope("root\\MyApp")
+        /// 
+        /// 'Explicit call to connect the scope object to the WMI namespace
+        /// s.Connect()
+        /// 
+        /// 'The following doesn't do any implicit scope connections because s is already connected.
+        /// Dim o As New ManagementObject(s, "Win32_LogicalDisk=""C:""", null)
+        ///    </code>
+        /// </example>
+        public void Connect ()
+        {
+            Initialize ();
+        }
+
+        internal void Initialize ()
+        {
+            //If the path is not set yet we can't do it
+            if (null == prvpath)
+                throw new InvalidOperationException();
+
+
+            /*
+             * If we're not connected yet, this is the time to do it... We lock
+             * the state to prevent 2 threads simultaneously doing the same
+             * connection. To avoid taking the lock unnecessarily we examine
+             * isConnected first
+             */ 
+            if (!IsConnected)
+            {
+#pragma warning disable CA2002
+                lock (this)
+#pragma warning restore CA2002
+                {
+                    if (!IsConnected)
+                    {
+                        // The locator cannot be marshalled accross apartments, so we must create the locator
+                        // and get the IWbemServices from an MTA thread
+                        if(!MTAHelper.IsNoContextMTA())
+                        {
+                            //
+                            // Ensure we are able to trap exceptions from worker thread.
+                            //
+                            ThreadDispatch disp = new ThreadDispatch ( new ThreadDispatch.ThreadWorkerMethodWithParam ( InitializeGuts ) ) ;
+                            disp.Parameter = this ;
+                            disp.Start ( ) ;
+                        }
+                        else
+                            InitializeGuts(this);
+                    }
+                }
+            }
+        }
+
+        void InitializeGuts(object o)
+        {
+            ManagementScope threadParam = (ManagementScope) o ;
+            IWbemLocator loc = (IWbemLocator) new WbemLocator();
+            IntPtr punk = IntPtr.Zero;
+
+            if (null == threadParam.options)
+            {
+                threadParam.Options = new ConnectionOptions();
+            }
+
+            string nsPath = threadParam.prvpath.GetNamespacePath((int)tag_WBEM_GET_TEXT_FLAGS.WBEMPATH_GET_SERVER_AND_NAMESPACE_ONLY);
+
+            // If no namespace specified, fill in the default one
+            if ((null == nsPath) || (0 == nsPath.Length))
+            {
+                // NB: we use a special method to set the namespace
+                // path here as we do NOT want to trigger an
+                // IdentifierChanged event as a result of this set
+                        
+              
```

Comment:
Check can be removed: .NET Core will always be running on higher than XP.

Reply:
Hmm.. this comment was meant for line 1131 below (where `Environment.OSVersion` is used). Not sure why GitHub placed the comment here. (Maybe an issue commenting on a large file from Safari on macOS?)